userbot/utils/clients/wakatime.py

```diff
@@ -0,0 +1,61 @@
+__all__ = [
+    "WakatimeClient",
+]
+
+from base64 import b64encode
+from datetime import datetime
+from math import floor
+
+from .base import BaseClient
+
+
+def format_time(seconds: int) -> str:
+    if seconds < 60:
+        return "no coding stats for today"
+
+    hours = int(seconds // 3600)
+    minutes = int((seconds - hours * 3600) // 60)
+
+    return "{0}{1}".format(f"{hours}h" if hours > 0 else "", f" {minutes}m" if minutes > 0 else "")
+
+
+class WakatimeClient(BaseClient, base_url="https://wakatime.com/api/v1/"):
+    def __init__(self, token: str):
+        super().__init__()
+        self._client.headers = {"Authorization": f"Basic {b64encode(token.encode()).decode()}"}
+
+    @staticmethod
+    def __process_stats_entity(entities) -> list[dict]:
+        return [
+            {"name": entity["name"], "text": entity["text"], "percent": entity["percent"]}
+            for entity in entities
+        ]
```

Comment:
1. Better use function in favor of staticmethod.
2. Use dataclasses instead of dict with fixed keys.